tools/prepare-source.sh

```diff
@@ -1,25 +1,29 @@
 #!/bin/sh
 set -eux
 
-GO_VERSION=1.22.0
+GO_VERSION=1.22.12
 OAPI_VERSION=2.4.1
-
-GO_BINARY=$(go env GOPATH)/bin/go$GO_VERSION
 TOOLS_PATH="$(realpath "$(dirname "$0")/bin")"
 
-# Install Go SDK
-go install golang.org/dl/go$GO_VERSION@latest
-$GO_BINARY download
+# Pin Go and toolbox versions at a reasonable version
+go get go@$GO_VERSION toolchain@$GO_VERSION
+
+# Update go.mod and go.sum:
+go mod tidy
 
 # Ensure dev tools are installed
-test -e "$TOOLS_PATH/goimports" || GOBIN=$TOOLS_PATH $GO_BINARY install golang.org/x/tools/cmd/goimports@latest
-("$TOOLS_PATH/oapi-codegen" -version | grep "$OAPI_VERSION" >/dev/null) || GOBIN=$TOOLS_PATH $GO_BINARY install github.com/oapi-codegen/oapi-codegen/v2/cmd/oapi-codegen@v$OAPI_VERSION
+test -e "$TOOLS_PATH/goimports" || GOBIN=$TOOLS_PATH go install golang.org/x/tools/cmd/goimports@latest
+("$TOOLS_PATH/oapi-codegen" -version | grep "$OAPI_VERSION" >/dev/null) || GOBIN=$TOOLS_PATH go install github.com/oapi-codegen/oapi-codegen/v2/cmd/oapi-codegen@v$OAPI_VERSION
 
-GOBIN=$TOOLS_PATH $GO_BINARY generate -x -mod=mod ./...
+# Generate source (skip vendor/):
+GOBIN=$TOOLS_PATH go generate -x ./cmd/... ./internal/...
```

Comment:
The `TOOLS_PATH` will disappear once we upgrade to 1.24 which features the all new Go tool pinning feature. In short, external tools will be added to `go.mod` now ensuring we all are on the same version. Until then, this is I believe a good workaround for the time being.